pinot-core/src/test/java/org/apache/pinot/queries/InterSegmentOrderByMultiValueQueriesTest.java

```diff
@@ -41,6 +42,43 @@ public void testGroupByOrderByMVSQLResults(String query, List<Object[]> expected
             expectedResults, expectedResults.size(), expectedDataSchema);
   }
 
+  /**
+   * Tests the in-segment trim option for GroupBy OrderBy query
+   */
+  @Test(dataProvider = "orderByDataProvider")
+  public void testGroupByOrderByMVTrimOptLowLimitSQLResults(String query, List<Object[]> expectedResults,
+      long expectedNumEntriesScannedPostFilter, DataSchema expectedDataSchema) {
+
+    expectedResults = expectedResults.subList(0, expectedResults.size() / 2);
```

Comment:
I don't see the value of this test. This test only changes the limit of the query

Reply:
This test aims to trigger the in-segment trim by providing a low limit number and a low trimSize, so the size of intermediateResult is greater than max(5*limit, trimSize). intermediateResult will be trimmed by the in-segment trim feature and then a limit will be applied (this is unavoidable). Still, we can test to see if everything works in the in-segment trim phase